resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsLifecycleManager.scala

```diff
@@ -166,15 +166,19 @@ private[spark] class ExecutorPodsLifecycleManager(
   private def removeExecutorFromK8s(execId: Long, updatedPod: Pod): Unit = {
     Utils.tryLogNonFatalError {
       if (shouldDeleteExecutors) {
+        // Get pod before deleting it, we can skip deleting if pod is already deleted so that
```

Comment:
How can I help you to verify this? I'm wondering if you can provide a test coverage.

Reply:
@dongjoon-hyun Thanks for your review, provided a test coverage